app/src/main/java/org/fossasia/pslab/communication/sensors/BMP180.java

```diff
@@ -0,0 +1,151 @@
+    private double x2 = pow(160, 2) * pow(2, -25) * readInt(0xB8);
+    private double y0 = c4 * pow(2, 15);
+    private double y1 = c4 * c3;
+    private double y2 = c4 * b1;
+    private double p0 = (3791.0 - 8.0) / 1600.0;
+    private double p1 = 1.0 - 7357.0 * pow(2, -20);
+    private double p2 = 3038.0 * 100.0 * pow(2, -36);
+    private double T = 25;
+    private double PR;
+    private double baseline;
+
+    private HashMap<String, ArrayList> params = new java.util.HashMap<>();
+
+
+    public BMP180(I2C i2c) throws IOException, InterruptedException
+    {
+        i2c = this.i2c;
+        Log.v("calib", (new double[]{c3, c4, b1, c5, c6, mc, md, x0, x1, x2, y0, y1, p0, p1, p2}).toString());
+        params.put("setOversampling",new ArrayList(Arrays.asList(0,1,2,3)));
+        initTemperature();
+        readTemperature();
+        initPressure();
+        baseline = readPressure();
+    }
+
+    int readInt(int addr)
+    {
+        return(Integer.parseInt(Integer.toString(addr).trim()));
+    }
+    double readUInt(int addr) throws IOException
+    {
+        double v;
+        ArrayList<Character> vals = i2c.readBulk(ADDRESS, addr, 2);
+        v = 1. * ((vals.get(0) << 8) | vals.get(1));
+        return v;
+    }
+
+    void initTemperature() throws IOException, InterruptedException {
+        i2c.writeBulk(ADDRESS, new int[]{REG_CONTROL, CMD_TEMP});
+        TimeUnit.MILLISECONDS.sleep(5);
+    }
+
+    Double readTemperature() throws IOException {
+        ArrayList<Character> vals = i2c.readBulk(ADDRESS, REG_RESULT, 2);
+        if (vals.size() == 2) {
+            T = (vals.get(0) << 8) + vals.get(1);
+            double a = c5 * (T - c6);
+            T = a + (mc / (a + md));
+            return T;
+        } else {
+            return null;
+        }
+    }
+
+     void setOversampling(int num)
+    {
+        oversampling = num;
+    }
+
+    void initPressure() throws IOException, InterruptedException
+    {
+        int[] os = {0x34, 0x74, 0xb4, 0xf4};
+        int [] delays = {5, 8, 14, 26};
+        i2c.writeBulk(ADDRESS, new int[] {REG_CONTROL, oversampling});
+        TimeUnit.MILLISECONDS.sleep(delays[oversampling]);
+
+    }
+
+    Double readPressure() throws IOException
+    {
+        ArrayList<Character> vals = i2c.readBulk(ADDRESS,REG_RESULT, 3);
+        if (vals.size() == 3)
+        {
+            double P = 1. * (vals.get(0) << 8) + vals.get(1) + (vals.get(2) / 256.0);
+            double s = T - 25.0;
+            double x = (x2 * pow(s, 2)) + (x1 * s) + x0;
+            double y = (y2 * pow(s, 2)) + (y1 * s) + y0;
+            double z = (P - x) / y;
+            PR = (p2 * pow(z, 2)) + (p1 * z) + p0;
+            return PR;
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+    double altitude()
+    {
+        // baseline pressure needs to be provided
+        return (44330.0 * (1 - pow( PR / baseline, 1 / 5.255)));
+    }
+
```

Comment:
Coding style doesn't match the style followed in entire repo, instead of (double P, double A) you can write (double pressure,double altitude) or something like that helps in understanding the code and concept better. ( If something isn't clear, put a comment )